### DISCUSSION TITLE
RPI 3B no sound via UGREEN sound card and Clippy EM272Z1

### DISCUSSION THREAD

**Discussion by prutsky** [Nachtzuster/BirdNET-Pi#248] [Q&A] [answered]
Hello,
I installed BirdNET-PI on a spare RPI and bought a [UGREEN usb soundadapter](https://www.amazon.nl/dp/B087T5H3MQ?ref=ppx_yo2ov_dt_b_fed_asin_title) and an mono [Clippy EM272Z1](https://www.veldshop.nl/nl/clippy-em272z1-mono-microphone.html) microphone. 

Somehow, with the sound card an microphone attached, BirdNET-Pi won't record anything. 

A browsed around the net and found some testing commands. So I did a `lsusb` and got the following response:
`birdnetpi:~/BirdNET-Pi$ lsusb

Bus 001 Device 004: ID 12d1:0010 Huawei Technologies Co., Ltd. KT USB Audio
Bus 001 Device 003: ID 0424:ec00 Microchip Technology, Inc. (formerly SMSC) SMSC9512/9514 Fast Ethernet Adapter
Bus 001 Device 002: ID 0424:9514 Microchip Technology, Inc. (formerly SMSC) SMC9514 Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub`

So I woulds say the _Huawei Technologies Co., Ltd. KT USB Audio_ is the UGreen sound card.

But when I do `arecord -l`
`birdnetpi:~/BirdNET-Pi$ arecord -l

**** List of CAPTURE Hardware Devices ****`

No hardware devices are returned. 

Checking with `alsamixer -m` I can set the KT USB audio. But when pressing F4 I get the message "This audio device does not have any capture controls".

What am I missing here?

**Comment by alexbelgium** [comment]
Hi, in don't have a definite answer but here are some thoughts.
- I've tried several DAC, and the ones that worked best for me where based on KT0210 chipsets like https://eu.ugreen.com/de/products/ugreen-usb-to-3-5mm-audio-jack-usb-a-sound-card-adapter. I also bought another ugreen DAC with another chipsets (ALC 5686 I think) and it was never recognized. Here I am not sure what you have : the amazon link says ALC4030, but the lsusb says KT usb. Anyway as long as appears, it should mean the DAC is correctly detected
- Your DAC only supports Trrs per the amazon page, but the veldshop.nl link does not mention if you got the TRRS or TRS version. I think it's the TRS that you have, because I bought this different article as being advertised as trrs : https://www.veldshop.nl/nl/smart-clippy-em272z1-mono-omni-microphone.html

Veldshop has a super reactive customer support, so I would recommend that you contact them to check! And then see which is the easiest to change between the DAC and microphone ;)

**Reply by prutsky** [reply]
Thans for the info! I'll check with Veldshop if my microphone also supports TRRS. 

Well, digging a bit more into this. My microphone only has 3 rings, so it will be a TRS version.

But when I read about compatibility I find [this link](https://www.vcelink.com/blogs/focus/trs-vs-trrs?srsltid=AfmBOor-4hvw9sUX0kQdPvJn8vQFHDKWJa1I8eW8Sk1SN8cAU6XPh2n2). 

Which says:
_**Compatibility**

Typically, TRS and TRRS connectors can be interchanged and plugged into their sockets, as long as their sizes match. For example, if you insert a TRS connector into a TRRS socket, the speaker function won't be available, but the microphone signal will be active on your device._

So I would say, this won't be a big problem? But [this link](https://github.com/mcguirepr89/BirdNET-Pi/wiki/Microphones-and-USB-audio-sound-cards-for-BirdNET-Pi) says you shouldn't interchange TRS with TRRS.

**Reply by alexbelgium** [reply, marked as answer]
Well I was more referring to this from the amazon page as a potential cause of why the DAC would show but not the microphone ...
![image](https://github.com/user-attachments/assets/be1e1432-ddb0-40cd-933c-4e4f6702517a)

I used the RODE AI mini as a DAC for some time and they had a specific note that it supported both TRRS and TRS microphone in a way that implied that it was a specific feature and not something given... I guess veldshop would have a more definite answer on that topic ;-)

Do you have another system that you could test it on? For example, if you have a windows computer you could see if it recognizes the microphone indicating a software related issue ; or if it is also not recognized indicating more an hardware one

**Reply by prutsky** [reply]
I tested it on a Windows machine. The audiocard is also recognised as a **ouput** (!) KT USB Audio device. But only as I connect the microphone.

I seems that I overlooked the stringent specs for the plug. And by unlucky change selected the wrong Clippy EM272 :(

I'll try to return and swap it for the correct TRRS version at Veldshop.

**Reply by alexbelgium** [reply]
Ah, sorry for you. I've also tested many systems before settling on mine... But ugreen / em272 should be an excellent combo! ugreen is more on the cost effective side, but other solutions are much more expensive. If possible you should get a deadcat too : it reduces the sound of wind and improves also the quality of signal.

**Reply by prutsky** [reply]
My deadkitten is now waiting for its replaced internals. 
![image](https://github.com/user-attachments/assets/52ba94bf-2f2c-4808-bdd5-be40e66b208a)
🤞🏻

**Comment by stevedee** [comment]
Just to point out that there are 2 wiring standards for TRRS (4 pole) jack plugs.

The sensible one is: OMTP where the screen of the cable connects to the shield of the plug.

The crazy one is: CTIA/AHJ  where the screen of the cable connects to the second ring of the plug.

With CTIA/AHJ the metal body of the jack is connected to the mic signal, so it can't be fully screened and will hum/buzz when you touch it.

If you need a metal bodied plug for best screening (e.g. transmitting equipment, low noise [mic] audio & so on) then OMTP is the best configuration.

If you are wiring/soldering a screened microphone cable to a 4 pole screen bodied jack plug for CTIA/AHJ, then 'good luck' with that.

If you want a specific brand of USB adapter (e.g. uGreen) then I recommend you get a model with 2 jack sockets; one for the mic and one for audio output, as these will always interface with TRS plugs, where the screen is connected to the body of the plug.
